src/models/onnxruntime_api.h

```diff
@@ -78,9 +85,72 @@ namespace Ort {
 /// Before using this C++ wrapper API, you MUST call Ort::InitApi to set the below 'api' variable
 inline const OrtApi* api{};
 inline void InitApi() {
+#if defined(__ANDROID__)
+  // If the GenAI library links against the onnxruntime library, it will have a dependency on a specific
+  // version of OrtGetApiBase.
+  //
+  // e.g. nm -D libonnxruntime-genai.so built this way shows it requires version 1.19.0 of OrtGetApiBase:
+  //  U OrtGetApiBase @VERS_1.19.0
+  //
+  //
+  // If you referenced those two packages in an app, and libonnxruntime-genai.so had a dependency on libonnxruntime.so
+  // for OrtGetApiBase, and the ORT versions don't match, you will get a cryptic runtime error like:
+  //   java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "OrtGetApiBase" referenced by
+  //                                   "/data/app/<appname>/base.apk!/lib/x86_64/libonnxruntime-genai.so"
+  //
+  // In order to be flexible we need to add complexity here by:
+  //   - don't call OrtGetApiBase directly so libonnxruntime-genai.so does not have a hard dependency
+  //     on libonnxruntime.so for the symbol.
+  //   - use dlopen/dysm to manually load the onnxruntime library and get the OrtGetApiBase function pointer.
+  //     - requires the Android app has referenced both the GenAI and ORT Android packages so the library is available.
+  //   - iterate between the current ORT version we're aware of, and a minimum required version, so that we work with
+  //     any libonnxruntime.so that supports one of those versions.
+  //
+
+  const std::string path = "libonnxruntime.so";  // "libonnxruntime4j_jni.so" is also an option if we have issues
+  __android_log_print(ANDROID_LOG_INFO, "GenAI", "Attempting to dlopen %s native library", path.c_str());
+
+  using OrtApiBaseFn = const OrtApiBase* (*)(void);
+  OrtApiBaseFn ort_api_base_fn = nullptr;
+
+  void* ort_lib_handle = dlopen(path.c_str(), RTLD_LOCAL);
+  if (ort_lib_handle == nullptr) {
+    __android_log_assert("ort_lib_handle != nullptr", "GenAI", "Failed to load %s", path.c_str());
```

Comment:
should we instead throw an exception to be consistent with the non-Android code path?

Reply:
This code executes in the context on a Java System.loadLibrary call and it's not clear to me how safe throwing a C++ exception in the genai library and hoping for it and its message to get back to that level with the message intact is.

The use of android log guarantees the message will be output, and the assert will kill the app, which I think is desirable for such a fundamental issue.

Reply:
logging and throwing is also a possibility.

regarding the terminate on error behavior, would that also be more appropriate for other platforms?

Reply:
In this specific case I'd prefer to completely die vs throwing and hoping the app that is using the library handles the exception well. It's a fundamental setup error for the ORT library to not be available so should only happen during development. That expected setup/behavior is however Android specific. 

As for other platforms I think we can consider the desired behavior when we separate out ORT from genai in those packages. Not sure what is available/useful. Android's __android_log_assert has the nice property of being debugger friendly (as below, and as I think this is a development-time issue that's very helpful). But doing something like an `abort` in c++ is generally horrible (immediate exit, hard to debug). 

```c
/*
 * Log an assertion failure and SIGTRAP the process to have a chance
 * to inspect it, if a debugger is attached. This uses the FATAL priority.
 */
void __android_log_assert(const char *cond, const char *tag, const char *fmt, ...)
```